### ISSUE TITLE
OBW - Remove unused OnboardingProducts and Purchase code 

### ISSUE THREAD

**Issue by crunnells** [woothemes/woocommerce#39955]
We’ve investigated an issue with the WCCOM search endpoint and identified a problem with the functionality of the [Purchase](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Purchase.php) task following the launch of the core profiler.

The task relies on data from the `product_type` field, which is no longer being collected as we have discontinued asking for product types. As a solution, we’ve decided to default to using physical as the product type.

However, when the physical is passed without product [key](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/Admin/Onboarding/OnboardingProducts.php#L32), it doesn’t yield any results, as indicated in this [line](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/Admin/Onboarding/OnboardingProducts.php#L123).

TL;DR — when the selected product type is set to physical (which is now the default after the core profiler launch), the Purchase task is not being displayed.

As a result, we've decided to remove the Purchase task in order to keep the codebase clean.

We are also looking to remove / deprecate the associated code ([REST API](https://github.com/woocommerce/woocommerce/blob/00496281afc93a18e533852f4b3d79a610c86939/plugins/woocommerce/src/Admin/API/OnboardingProductTypes.php#L20), [OnboardingProducts class](https://github.com/woocommerce/woocommerce/blob/00496281afc93a18e533852f4b3d79a610c86939/plugins/woocommerce/src/Internal/Admin/Onboarding/OnboardingProducts.php#L16-L17))

**Comment by ilyasfoo** [comment]
Hey, @crunnells! Are you working on this? Otherwise, I can pick it up this week since I'm the porter

**Comment by crunnells** [comment]
Hey @ilyasfoo there's a PR that just removes the reference from the default lists: #40121 but talking to @adrianduffell, he suggested we deprecate the Purchase task for sites that still have that item in their task list.

**Comment by ilyasfoo** [comment]
> we deprecate the Purchase task for sites that still have that item in their task list.

@crunnells @adrianduffell I think that may be moot because the only way the purchase task gets displayed is by choosing a theme or a product (subscription, etc) in the "product type" step in the old wizard, which is no longer accessible. I think removing the purchase task outright is the correct way forward here.

**Comment by adrianduffell** [comment]
@crunnells @ilyasfoo Yep I'd agree on removing the purchase task outright. There will be some stores still displaying it due to previous usage in the old wizard, but I think enough time has passed now that they are unlikely to ever complete the task.

Just to clarify the deprecations -- I think we'd need to deprecate the [REST API endpoint](https://github.com/woocommerce/woocommerce/blob/00496281afc93a18e533852f4b3d79a610c86939/plugins/woocommerce/src/Admin/API/OnboardingProductTypes.php#L20) and maybe also the  [OnboardingProducts class](https://github.com/woocommerce/woocommerce/blob/00496281afc93a18e533852f4b3d79a610c86939/plugins/woocommerce/src/Internal/Admin/Onboarding/OnboardingProducts.php#L16-L17), but this is less urgent.

**Comment by ilyasfoo** [comment]
> Just to clarify the deprecations -- I think we'd need to deprecate the [REST API endpoint](https://github.com/woocommerce/woocommerce/blob/00496281afc93a18e533852f4b3d79a610c86939/plugins/woocommerce/src/Admin/API/OnboardingProductTypes.php#L20) and maybe also the [OnboardingProducts class](https://github.com/woocommerce/woocommerce/blob/00496281afc93a18e533852f4b3d79a610c86939/plugins/woocommerce/src/Internal/Admin/Onboarding/OnboardingProducts.php#L16-L17), but this is less urgent.

Ah, gotcha. Makes sense.
